docs/interface/Metadata.md

```diff
@@ -1,50 +1,94 @@
-- `Metadata::Node`
+The following fields specify the topology of the variable, and are
+exclusive:
 
-These fields specify what index space the variable is allocated over. E.g.
-`Metadata::Cell` specifies that the variable is allocated over the cell index
-space in each block.
+- `Metadata::None`: no topology specified. The variable could be
+  anywhere, or location is not a meaningful concept for this variable.
+- `Metadata::Cell`: The variable is defined on cell centers. The
+  variable is likely volume-averaged.
+- `Metadata::Face`: The variable is defined on cell faces. The
+  variable is likely area-averaged.
+- `Metadata::Edge`: The variable is defined on cell edges. The
+  variable is likely length-averaged.
+- `Metadata::Node`: The variable is defined at nodes, i.e.,
+  cell-corners. The variable might be volume-averaged, or defined
+  pointwise.
 
 ### Variable Behaviors
-TODO
-- `Metadata::Advected`
-- `Metadata::Conserved`
-- `Metadata::Intensive`
+
+These flags can be used to tell an application code how to treat a
+variable in relation to the problem.
+
+- `Metadata::Advected` implies a variable is advected with the flow of
+  another variable, e.g., a velocity.
+- `Metadata::Conserved` implies a variable obeys a conservation law.
+- `Metadata::Intensive` implies that the value of a variable does not
+  scale with volume.
 
 ### Output
-TODO
-- `Metadata::Restart`
-- `Metadata::Graphics`
 
-### Tensors
-TODO
-- `Metadata::Vector`
-- `Metadata::Tensor`
+These flags specify how a variable interacts with I/O. Enable them to
+enable output properties.
+
+- `Metadata::Restart` implies a variable is required in restart files
+- `Metadata::Graphics` implies a varaible should be output for visualization
+
+### Tensor properties and boundaries
+
+For multidimensional variables, these flags specify how to treat the
+individual components at boundaries. For concreteness, we will discuss
+reflecting boundaries. But this may apply more broadly. A variable
+with no flag set is assumed to be a *Scalar*. Scalars obey 
+[Dirichlet boundary conditions](https://en.wikipedia.org/wiki/Dirichlet_boundary_condition)
+at reflecting boundaries and are set to a constant value.
+
+- `Metadata::Vector` implies the variable transforms as a *vector* at
+  reflecting boundaries. And so i-th component is flipped for a
+  boundary in the i-th direction.
+- `Metadata::Tensor` is the generalization of the vector boundary
+  condition, but for tensor quantities.
 
 ### Independent/Derived
-TODO
-- `Metadata::Independent`
-- `Metadata::Derived`
+
+These flags specify to an application code, and the infrastructure,
+whether or not a variable is part of independent state. Derived
+quantities can be calculated from the set of independent quantities,
+while independent quantities cannot.
+
+- `Metadata::Independent` implies the variable is part of independent
+  state
+- `Metadata::Derived` implies the variable can be calculated, given
+  the independent state
```

Comment:
These are mutually exclusive, right?

Reply:
Is every variable either independent or derived, or could it be neither?

Reply:
These are mutually exclusive and everything should be either independent or derived. I will specify that.